packages/amplify-graphql-transformer-interfaces/src/transform-host-provider.ts

```diff
@@ -31,7 +32,7 @@ export interface DynamoDbDataSourceOptions extends DataSourceOptions {
 export interface TransformHostProvider {
   setAPI(api: GraphqlApiBase): void;
 
-  addHttpDataSource(name: string, endpoint: string, options?: DataSourceOptions, scope?: Construct): HttpDataSource;
+  addHttpDataSource(name: string, endpoint: string, options?: HttpDataSourceOptions, scope?: Construct): HttpDataSource;
```

Comment:
This was necessary to pass the authorizationConfig (signing region / service name) to the HTTP data source on creation. [[Docs reference](https://docs.aws.amazon.com/appsync/latest/APIReference/API_AuthorizationConfig.html)]

Reply:
Please confirm this doesn't break anything in the gen2 schema builder HTTP data source flow https://docs.amplify.aws/javascript/build-a-backend/data/custom-business-logic/connect-http-datasource/#step-2---add-your-rest-api-or-http-api-as-datasource

Reply:
Looks like the answer is no, it doesn't break anything.
[`AmplifyGraphqlAPI.addHttpDataSource`](https://github.com/aws-amplify/amplify-category-api/blob/a1d16203428d6d2ea3b2ee048b310fdef1b24c60/packages/amplify-graphql-api-construct/src/amplify-graphql-api.ts#L319-L321) already takes a `HttpDataSourceOptions`.